mu-plugins/pwcc-helpers/inc/tachyon-mods/namespace.php

```diff
@@ -0,0 +1,463 @@
+			'height' => $h,
+			'file' => "{$filename}{$crop_hash}-{$w}x{$h}.{$ext}",
+			'mime-type' => $mime_type,
+		];
+	}
+
+	return $data;
+}
+
+/**
+ * Tachyon gravity string can sometimes be reversed.
+ *
+ * Gravity is sometimes reported as east/west before north/south.
+ * This causes problems with the service as `eastnorth` is not recognised.
+ *
+ * @param $tachyon_args array Arguments for calling Tachyon.
+ * @param $image        array The image details.
+ *
+ * @return array Modified arguments with gravity corrected.
+ */
+function filter_tachyon_gravity( $tachyon_args, $image ) {
+	if ( ! is_array( $tachyon_args ) || ! isset( $tachyon_args['gravity'] ) ) {
+		return $tachyon_args;
+	}
+
+	$gravity = $tachyon_args['gravity'];
+
+	// Ensure Gravity is not empty.
+	if ( $gravity === '' ) {
+		unset( $tachyon_args['gravity'] );
+		return $tachyon_args;
+	}
+
+	// Check if too short to need modification.
+	if ( strlen( $gravity ) <= 5 ) {
+		return $tachyon_args;
+	}
+
+	// Ensure direction is correct: northeast instead of eastnorth.
+	$gravity_tail = substr( $gravity, -5 );
+	if ( $gravity_tail === 'north' || $gravity_tail === 'south' ) {
+		$gravity = $gravity_tail . substr( $gravity, 0, -5 );
+	}
+
+	// Normalise for orientation.
+	$image_id  = $image['attachment_id'];
+	$image_meta  = wp_get_attachment_metadata( $image_id );
+
+	$orientation = '';
+	if ( isset( $image_meta['height'], $image_meta['width'] ) ) {
+		$orientation = ( $image_meta['height'] > $image_meta['width'] ) ? 'portrait' : 'landscape';
+	}
+
+	switch ( $orientation ) {
+		case 'portrait':
+			if ( strpos( $gravity, 'north' ) !== false ) {
+				$gravity = 'north';
+			} elseif ( strpos( $gravity, 'south' ) !== false ) {
+				$gravity = 'south';
+			}
+			break;
+		case 'landscape':
+			if ( strpos( $gravity, 'east' ) !== false ) {
+				$gravity = 'east';
+			} elseif ( strpos( $gravity, 'west' ) !== false ) {
+				$gravity = 'west';
+			}
+			break;
+	}
+
+	$tachyon_args['gravity'] = $gravity;
+	return $tachyon_args;
+}
+
+/**
+ * Filters 'img' elements in post content to add 'srcset' and 'sizes' attributes.
+ *
+ * @param string $content The raw post content to be filtered.
+ *
+ * @return string Converted content with 'srcset' and 'sizes' attributes added to images.
+ */
+function make_content_images_responsive( $content ) {
+	$images = \Tachyon::parse_images_from_html( $content );
+
+	if ( empty( $images ) ) {
+		// No images, leave early.
+		return $content;
+	}
+
+	// This bit is from Core.
+	$selected_images = [];
+	$attachment_ids = [];
+	foreach ( $images['img_tag'] as $image ) {
+		if ( false === strpos( $image, ' srcset=' ) && preg_match( '/wp-image-([0-9]+)/i', $image, $class_id ) && absint( $class_id[1] ) ) {
+			$attachment_id = $class_id[1];
+			/*
+			 * If exactly the same image tag is used more than once, overwrite it.
+			 * All identical tags will be replaced later with 'str_replace()'.
+			 */
+			$selected_images[ $image ] = $attachment_id;
```

Comment:
Set this up with `$image_data` instead of below. That prevent multiple passes of the same image tag.